wild/tests/sources/trivial-dynamic.c

```diff
@@ -15,6 +15,8 @@
 
 //#Config:origin:default
 //#LinkArgs:-z now -z origin
+// Some versions of GNU ld don't use the DT_FLAGS at all and rely on DT_FLAGS_1 and tags.
+//#DiffIgnore:.dynamic.DT_FLAGS.ORIGIN
```

Comment:
Alternatively, linker-diff could be taught to consider  `.dynamic.DT_FLAGS.ORIGIN` and `.dynamic.DT_FLAGS_1.ORIGIN` equivalent.

Reply:
If it's something that just differs based on target arch, then I'd say that we should try to match what GNU ld does on that target arch, since it's possible that glibc on that arch depends on that behaviour. However, if it varies between versions of GNU ld on the same target arch, then making linker-diff treat them as equivalent sounds good.

I guess the other option would be to set them all on all architectures, then treat them as the same in linker-diff. That does have the appeal of simplicity to it.

Reply:
I'm afraid there is no clear preference here.
Binutils 2.43 target AArch64 on Arch Linux uses tags instead of flags, but 2.43-2.44 on Ubuntu/openSUSE still use flags (can be seen in the failed build).

Arch's configuration is very simple but doesn't enable new-dtags:
https://gitlab.archlinux.org/archlinux/packaging/packages/aarch64-linux-gnu-binutils/-/blob/main/PKGBUILD?ref_type=heads#L33
Maybe that's the reason, I'll test that.